src/storage/page_buffer.c

```diff
@@ -7914,6 +7930,11 @@ pgbuf_victimize_bcb (THREAD_ENTRY * thread_p, PGBUF_BCB * bufptr)
     }
   assert (bufptr->latch_mode == PGBUF_NO_LATCH);
 
+  if (bufptr->iopage_buffer->iopage.prv.lsa > pgbuf_Pool.get_highest_evicted_lsa ())
+    {
+      pgbuf_Pool.set_highest_evicted_lsa (bufptr->iopage_buffer->iopage.prv.lsa);
+    }
```

Comment:
This is not entirely safe. Concurrent checks on `bufptr->iopage_buffer->iopage.prv.lsa > pgbuf_Pool.get_highest_evicted_lsa ()` may end up setting an LSA less than highest:

  - Two concurrent evictions, thread1 evicting a page with lsa1 and thread2 evicting a page with lsa2, both higher than current highest evicted LSA, but with lsa1 < lsa2.
  - thread2 calls set_highest_evicted_lsa first, setting lsa2 as highest evicted LSA
  - thread1 calls set_highest_evicted_lsa second, setting lsa1 as highest evicted LSA
  - highest evicted LSA is now lsa1, but it should be lsa2.

 Better to add a pgbuf_Pool function that checks when to update highest evicted lsa with compare and swap:

```c
void
pgbuf_buffer_pool::update_highest_evicted_lsa (log_lsa evicted_lsa)
{
  do
    {
      log_lsa crt_highest = highest_evicted_lsa;
      if (crt_highest > evicted_lsa)
        {
          // already higher
          break;
        }
    }
  while (!highest_evicted_lsa.compare_exchange_weak (crt_highest, evicted_lsa))
}
```